main/src/ca/uwaterloo/flix/api/Bootstrap.scala

```diff
@@ -467,9 +467,10 @@ class Bootstrap(val projectPath: Path, apiKey: Option[String]) {
     // Add sources and packages.
     reconfigureFlix(flix)
 
-    flix.check() match {
-      case Validation.Success(_) => Validation.success(())
-      case failure => Validation.toHardFailure(BootstrapError.GeneralError(flix.mkMessages(failure.errors)))
+    flix.check().toResult match {
+      case Result.Ok((_, Nil)) => Validation.success(())
+      case Result.Ok((_, errors)) => Validation.toHardFailure(BootstrapError.GeneralError(flix.mkMessages(errors)))
+      case Result.Err(errors) => Validation.toHardFailure(BootstrapError.GeneralError(flix.mkMessages(errors)))
     }
```

Comment:
This should be possible without converting to `Result` first. Maybe something like a `mapErr` or `mapFailure` could do the trick?

Reply:
Alright, I'll see if there are any more uses